src/core/utils/internal/isObject.ts

```diff
@@ -1,6 +1,6 @@
 /**
  * Determines if the given value is an object.
  */
-export function isObject(value: any): boolean {
+export function isObject(value: any): value is Record<string, any> {
```

Comment:
fwiw, this isn't exactly type-safe in a way you'd assume: [TS playground](https://www.typescriptlang.org/play/?ts=5.9.0-dev.20250601#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXywGcB5AIwCtwMAKANygmRAC54pUBPASlfsYSLwASuBwxgAHkIYYWVAHMANG04A+ANwBYAFA7QkWAgggM8ALZQOpEAEEYMALKXrZcqwDeSHDlbTZC+ABfeAAfeFRkM2sYAG0AXS1dbSxEeGoiVypqCytbeydc1y4ueHcdeHNnPMcq10SKgHoG+AA9AH4dYJAIQgQy7Qqc6zsawop6+CbWju1AnSA)